includes/class-amp-story-post-type.php

```diff
@@ -1500,4 +1512,29 @@ public static function change_embed_iframe_attributes( $output, $post ) {
 			$output
 		);
 	}
+
+	/**
+	 * Adds a new max image size to the images sizes available, and removes 'full'.
+	 *
+	 * In the AMP story editor, when selecting Background Media,
+	 * it will use this custom image size.
+	 * This will also be available for image blocks.
+	 * But the new size will only be available if the original image was at least as big as it is.
+	 *
+	 * @param array $image_sizes {
+	 *     An associative array of image sizes.
+	 *
+	 *     @type string $slug Image size slug, like 'medium'.
+	 *     @type string $name Image size name, like 'Medium'.
+	 * }
+	 * @return array $image_sizes The filtered image sizes.
+	 */
+	public static function add_new_max_image_size( $image_sizes ) {
+		if ( self::POST_TYPE_SLUG === get_post_type() ) {
```

Comment:
This is not working for me. The Ajax request that obtains the media library results does not include a post as context, as far as I can see. So when I insert the background image into the story page, I do not have the expected image size. I do, however, if I make this change:

```diff
--- a/includes/class-amp-story-post-type.php
+++ b/includes/class-amp-story-post-type.php
@@ -1530,11 +1530,7 @@ class AMP_Story_Post_Type {
 	 * @return array $image_sizes The filtered image sizes.
 	 */
 	public static function add_new_max_image_size( $image_sizes ) {
-		if ( self::POST_TYPE_SLUG === get_post_type() ) {
-			unset( $image_sizes['full'] );
-			$image_sizes[ self::MAX_IMAGE_SIZE_SLUG ] = __( 'Story Background', 'amp' );
-		}
-
+		$image_sizes[ self::MAX_IMAGE_SIZE_SLUG ] = __( 'Story Background', 'amp' );
 		return $image_sizes;
 	}
 }
```

Reply:
It's not working in that the full size image is being used, as opposed to an image filename ending in `-1080x1440.jpg`.

Reply:
Interesting, thanks for testing this. 

What was the original size of the image that you selected as the 'Background Media'?

These steps seem to work well locally:
1. In 'Background Media,' click 'Select Media' or 'Edit Media' 
2. Upload a [1900 x 1900 image](http://lorempixel.com/1900/1900/technics/)
3. Click 'Select'
4. Observe that the URL of the background image ends in 1440-1440.jpeg, meaning it used the `amp_story_page` size:

```
https://loc.test/wp-content/uploads/2019/06/story-1900-1900-3-1440x1440.jpeg
```
<img width="960" alt="bck-image-here" src="https://user-images.githubusercontent.com/4063887/59172142-15049100-8b0c-11e9-862a-385684c147a7.png">

These 4 steps also worked with these image sizes:
[1000 x 1500](http://lorempixel.com/1000/1500/technics/)
[1200 x 1500](http://lorempixel.com/1200/1500/technics/)
[1500 x 1600](http://lorempixel.com/1500/1600/technics/)
[1800 x 1800](http://lorempixel.com/1800/1800/technics/)

Of course, this didn't work with images less than 1440 in height, as they're not big enough to have an `amp_story_page` sized image (99999 x 1440).